language-design.md

```diff
@@ -16,19 +14,29 @@ are:
++ "The development environment suffers from literally decades of accretion of stupid hacks from a community containing, to a first-order approximation, zero software engineers." (Smith, 2016)
+  + This quotation from the additional reading on the R programming language is included because it was very reminiscent of a discussion with Professor Ben, in which we were discussing the separation of domain knowledge and programming language design / software engineering experience.
+  + This highlights the importance of the work we did at the Hive in learning how to interview domain experts. Since these are the individuals with the fluency in the domain, it is essential that software developers creating tools for them are able to empathize with their needs so that the language also has fluency. The other direction is also important (the lack of which inspired this quote about R). When domain specialists are alone in developing a language, the lack of designers and software engineers may lead to unintuitive solutions or software which is difficult to extend.
 
 ---
 
 **Question**
 
-How would you know a well-designed language? What are the symptoms? How would
-you know a poorly designed language? What are the symptoms?
+How would you know a well-designed language? What are the symptoms? How would you know a poorly designed language? What are the symptoms?
 
 **Response**
+A well-designed language should have a syntax that is intuitive enough that users can write programs accurately as well as a predictable grammar; Pavlus in his article remarks that Quorum, an evidence-based programming language, lets users write programs more accurately than in Perl. One thing to keep in mind is that a language that is intuitive to someone might not be to another. For example, AppleScript is supposedly intuitive to read, but certainly not more intuitive to write in for people who have been programming in C++ or Python. So it is more intuitive to write for people with less experience in traditional programming.
+
+A well-designed language should also have flexibility, the ability to grow. Steele states in his paper that [he] need[s] to design a language that can grow. [He] need[s] to plan ways in which it might grow - but [he] need[s], too, to leave some choices so that other persons can make those choices at a later time." (Steele, 1998).
 
+In addition, a well-designed language should fail early. This applies when designing a good API as well; According to Bloch, "the sooner you report a bug, the less damage it will do. Compile-time is best" (Bloch, 2006).
 
+Furthermore, a well-designed language should be easy to maintain. This is feasible only if there's a good documentation for the language. 
```

Comment:
What are some concrete markers of a poorly designed language? While I can assume the opposite of your points about well-designed languages making something a poor language, are there any definitive markers of a poor design or some contextual properties?